components/badge.css

```diff
@@ -0,0 +1,146 @@
+.badge-container {
+	position: relative;
+	width: fit-content;
+	padding: 0.5rem;
+}
+
+.status-badge {
+	position: absolute;
+	height: 1.2rem;
+	min-width: 1.2rem;
+	padding: 2px;
+	border-radius: 50%;
+	border: 2px solid white;
+}
+
+.badge-container .avatar-lg~.status-badge {
+	right: 1.8em;
+	bottom: 1.2em;
+	height: 1.6rem;
+	min-width: 1.6rem;
+}
+
+.badge-container .avatar-md~.status-badge {
+	right: 1.4em;
+	bottom: 1.2em;
+}
+
+.badge-container .avatar-sm~.status-badge {
+	right: 1.2em;
+	bottom: 0.6em;
+	height: 1rem;
+	min-width: 1rem;
+}
+
+.badge-container .avatar-xsm~.status-badge {
+	right: 1.4em;
+	bottom: 0.6em;
+	height: 0.7rem;
+	min-width: 0.7rem;
+}
+
+.status-offline {
+	background-color: var(--error-text);
+}
+
+.status-online {
+	background-color: var(--success-text);
+}
+
+.status-away {
+	background-color: var(--warning-text);
+}
+
+.badge-text {
+	padding: 0.3rem 0.2rem;
+}
+
+.badge-on-text,
+.badge-on-icon,
+.badge-on-btn {
+	position: absolute;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	color: var(--text-white);
+	background-color: var(--primary-color);
+	min-height: 1.3em;
+	min-width: 1.3em;
+	font-weight: 400;
+}
+
+.badge-on-text {
+	top: -0.2em;
+	right: -0.7em;
+	border-radius: 50%;
+}
+
+.badge-round {
+	border-radius: 50%;
+}
+
+.badge-square {
+	border-radius: 4px;
+	padding: 0 0.2em;
+}
+
+.badge-on-icon {
+	font-size: 0.8em;
+	top: 0.2em;
+	right: 0.3em;
+	padding: 0.2em 0.3em;
+	border-radius: 20px;
+}
+
+.badge-on-btn {
+	font-size: 0.9em;
+	top: 0.3em;
+	right: 0.5em;
+	border-radius: 50%;
+}
+
+.btn-badge-cont button {
+	font-size: 1rem;
+	width: 6.25rem;
+	padding: 0.37rem 1rem;
+	border-radius: 10px;
+	border: none;
+	background-color: var(--primary-color);
+	color: var(--text-white);
+	cursor: pointer;
+	margin: 0.5rem;
+}
+
+.btn-badge {
+	padding: 0.15rem 0.4rem;
+	color: var(--text-white);
+	border-radius: 50%;
+	font-size: 0.8rem;
+	background-color: var(--badge-red);
+	position: absolute;
+	left: 0.9rem;
+}
+
+.btn-toggle-cont button {
```

Comment:
don't use descendant selector, give the button a specific class

Reply:
Sure Buddy will do it!